### PR TITLE
[MithrilJS] Added censor and JSX children

### PR DESCRIPTION
**Pull request by maicol07**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://mithril.js.org/censor.html
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by maicol07** [comment]
@dead-claudia @spacejack I think the last error is a false positive, since in my manual tests everything worked well

**Comment by dead-claudia** [comment]
Error's legitimate, but not for this package.

**Review by dead-claudia — APPROVED** [review]

**Review by dead-claudia — CHANGES_REQUESTED** [review]
Error's legitimate. Please adjust the TS versions here, or alternatively rebase your PR to remove the diff.

File: types/mithril/index.d.ts [hunk 1]
```diff
@@ -3,7 +3,8 @@
 // Definitions by: Mike Linkovich <https://github.com/spacejack>
 //                 Claudia Meadows <https://github.com/dead-claudia>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.2
+// TypeScript Version: 5.0
+// Minimum TypeScript Version: 5.3
```
Comment (dead-claudia) [inline types/mithril/index.d.ts:7]: This is where the error's coming from, BTW.

Comment (maicol07) [reply types/mithril/index.d.ts]: Fixed! Didn't know that counts too!

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## mithril ([<kbd>unpkg</kbd>](https://unpkg.com/browse/mithril@latest/))
was missing the following properties:
1. m
1. Fragment
1. vnode
1. domFor
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0cd6a793e2fa8a5d8908c0d0e71718215d398837
</p>

File: types/mithril-global/index.d.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
-// Type definitions for non-npm package Mithril 2.0
+// Type definitions for non-npm package Mithril 2.2
 // Project: https://mithril.js.org/
 // Definitions by: Mike Linkovich <https://github.com/spacejack>
 //                 Claudia Meadows <https://github.com/dead-claudia>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.2
+// TypeScript Version: 5.3
```
Comment (sheetalkamat) [inline types/mithril-global/index.d.ts:6]:
```suggestion
```

File: types/mithril/index.d.ts [hunk 3]
```diff
@@ -1,9 +1,9 @@
-// Type definitions for Mithril 2.0
+// Type definitions for Mithril 2.2
 // Project: https://mithril.js.org/, https://github.com/mithriljs/mithril.js
 // Definitions by: Mike Linkovich <https://github.com/spacejack>
 //                 Claudia Meadows <https://github.com/dead-claudia>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 3.2
+// TypeScript Version: 5.3
```
Comment (sheetalkamat) [inline types/mithril/index.d.ts:6]:
```suggestion
```

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Comment by maicol07** [comment]
@sheetalkamat I had to revert your suggestions since TS 5.3 is required to not give errors in tests

File: types/mithril-global/index.d.ts (hunk 2 above)
Comment (sheetalkamat) [reply types/mithril-global/index.d.ts:6]: This is moving everyone using this package to typescript version 5.3 which is not recommended. 
Please resolve errors such that its not TS version dependent and remove this line.

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/mithril-global/index.d.ts (hunk 2 above)
Comment (maicol07) [reply types/mithril-global/index.d.ts]: Found a way to do this! If tests don't fail we're good to go!

**Review by dead-claudia — APPROVED** [review]

**Review by sheetalkamat — APPROVED** [review]

**Comment by maicol07** [comment]
Ready to merge

**Pull request merged**